core/cmd/local_client.go

```diff
@@ -43,14 +44,126 @@ import (
 // ownerPermsMask are the file permission bits reserved for owner.
 const ownerPermsMask = os.FileMode(0700)
 
+func openDB(cfg config.GeneralConfig, lggr logger.Logger) (db *sqlx.DB, err error) {
+	uri := cfg.DatabaseURL()
+	appid := cfg.AppID()
+	static.SetConsumerName(&uri, "App", &appid)
+	dialect := cfg.GetDatabaseDialectConfiguredOrDefault()
+	db, err = pg.NewConnection(uri.String(), string(dialect), pg.Config{
+		Logger:       lggr,
+		MaxOpenConns: cfg.ORMMaxOpenConns(),
+		MaxIdleConns: cfg.ORMMaxIdleConns(),
+	})
+	err = errors.Wrap(err, "failed to open db")
+	return
+}
+
+func applicationLockDB(cfg config.GeneralConfig, db *sqlx.DB, lggr logger.Logger, sig shutdown.Signal) (cleanup func(), err error) {
+	lggr.Debugf("Using database locking mode: %s", cfg.DatabaseLockingMode())
+
+	cleanup = func() {}
+
+	// Lease will be explicitly released on application stop
+	// Take the lease before any other DB operations
+	var leaseLock pg.LeaseLock
+	if cfg.DatabaseLockingMode() == "lease" || cfg.DatabaseLockingMode() == "dual" {
+		leaseLock = pg.NewLeaseLock(db, cfg.AppID(), lggr, cfg.LeaseLockRefreshInterval(), cfg.LeaseLockDuration())
+		if err = leaseLock.TakeAndHold(sig); err != nil {
+			return cleanup, errors.Wrap(err, "failed to take initial lease on database")
+		}
+		cleanup = leaseLock.Release
+	}
+
+	// Try to acquire an advisory lock to prevent multiple nodes starting at the same time
+	var advisoryLock pg.Locker
+	if cfg.DatabaseLockingMode() == "advisorylock" || cfg.DatabaseLockingMode() == "dual" {
```

Comment:
Could DRY this up a bit:
```suggestion
	lockingMode := cfg.DatabaseLockingMode()
	lggr.Debugf("Using database locking mode: %s", lockingMode)

	cleanup = func() {}

	// Lease will be explicitly released on application stop
	// Take the lease before any other DB operations
	var leaseLock pg.LeaseLock
	if lockingMode == "lease" || lockingMode == "dual" {
		leaseLock = pg.NewLeaseLock(db, cfg.AppID(), lggr, cfg.LeaseLockRefreshInterval(), cfg.LeaseLockDuration())
		if err = leaseLock.TakeAndHold(sig); err != nil {
			return cleanup, errors.Wrap(err, "failed to take initial lease on database")
		}
		cleanup = leaseLock.Release
	}

	// Try to acquire an advisory lock to prevent multiple nodes starting at the same time
	var advisoryLock pg.Locker
	if lockingMode == "advisorylock" || lockingMode == "dual" {
```